impl/src/planning/subplanners.rs

```diff
@@ -439,225 +426,208 @@ impl<'planner> CrateSubplanner<'planner> {
-        }
-        // Only add aliased normal deps to the Vec
-        if let Some(alias_pair) = aliased_dep_names.get(&dep_package.name) {
-          // Check whether the package's version matches the semver requirement
-          // of the package that had an alias
-          if alias_pair.0.matches(&dep_package.version) {
-            dep_set.aliased_deps.push(DependencyAlias {
-              target: buildable_target.clone(),
-              alias: (&alias_pair.1).replace("-", "_"),
-            })
-          }
-        }
+        let mut dep_set = dep_production.entry(platform_target).or_default();
+        self.process_dep(&mut dep_set, &dep.name, dep_kind, &dep_package)?
       }
     }
 
-    dep_set.aliased_deps.sort();
-    dep_set.build_deps.sort();
-    dep_set.build_proc_macro_deps.sort();
-    dep_set.dev_deps.sort();
-    dep_set.normal_deps.sort();
-    dep_set.proc_macro_deps.sort();
+    for set in dep_production.values_mut() {
+      set.build_proc_macro_dependencies.sort();
+      set.build_dependencies.sort();
+      set.dev_dependencies.sort();
+      set.proc_macro_dependencies.sort();
+      set.dependencies.sort();
+    }
 
-    Ok(dep_set)
+    Ok(dep_production)
   }
 
-  /// Generates the set of dependencies for the contained crate.
-  fn produce_deps(&self) -> Result<(DependencySet, Vec<TargetedDependencySet>)> {
-    let (default_deps, targeted_deps) = self.identify_named_deps()?;
-
-    let targeted_set = targeted_deps
-      .iter()
-      .map(|(target, deps)| TargetedDependencySet {
-        target: target.clone(),
-        dependencies: self._produce_deps(deps).unwrap(),
-      })
-      .collect::<Vec<TargetedDependencySet>>();
-
-    Ok((self._produce_deps(&default_deps)?, targeted_set))
-  }
+  fn process_dep(
+    &self,
+    dep_set: &mut CrateDependencyContext,
+    name: &str,
+    dep: &DepKindInfo,
+    pkg: &Package,
+  ) -> Result<()> {
+    let is_proc_macro = self.is_proc_macro(pkg);
+    let is_sys_crate = pkg.name.ends_with("-sys");
+
+    let build_dep = BuildableDependency {
+      name: pkg.name.clone(),
+      version: pkg.version.clone(),
+      buildable_target: self.buildable_target_for_dep(pkg)?,
+      is_proc_macro,
+    };
 
-  /// Yields the list of dependencies as described by the manifest (without version).
-  fn identify_named_deps(&self) -> Result<(DependencyNames, HashMap<String, DependencyNames>)> {
-    // Resolve dependencies into types
-    let mut default_dep_names = DependencyNames {
-      build_dep_names: Vec::new(),
-      dev_dep_names: Vec::new(),
-      normal_dep_names: Vec::new(),
-      aliased_dep_names: HashMap::new(),
+    use DependencyKind::*;
+    match dep.kind {
+      Build if is_proc_macro => dep_set.build_proc_macro_dependencies.push(build_dep),
+      Build => dep_set.build_dependencies.push(build_dep),
+      Development => dep_set.dev_dependencies.push(build_dep),
+      // TODO(GregBowyer): Is a proc-macro -sys crate a thing?
```

Comment:
Since writing this, was did you ever find out? I don't think it's a thing but that's likely a result of my limited experience 😅

Reply:
I have never seen one in the wild although `-sys` crates are by convention.

I will pull it, I think if it happens it will be a curious bug in the future to figure out (even if such a thing existed would it be a host or target compile for instance)